### PR TITLE
add clear error message when fixture is not defined (closes #2913)

### LINKED ISSUE #2913: An error appears when test starts without defined fixture
I missed `fixture` keyword and got
`TypeError: Cannot read property 'requestHooks' of null
at new requestHooks (D:\projects\testcafe\src\api\structure\test.js:16:42)`

### PR DESCRIPTION
**Pull request by AlexKamaev**

### CODE REVIEW

File: src/api/structure/test.js [hunk 1]
```diff
@@ -13,17 +13,19 @@ export default class Test extends TestingUnit {
...
     _add (name, fn) {
         assertType(is.string, 'apiOrigin', 'The test name', name);
         assertType(is.function, 'apiOrigin', 'The test body', fn);
+        assertType(is.nonNullObject, 'apiOrigin', `The fixture of '${name}' test`, this.fixture);
 
-        this.name = name;
-        this.fn   = wrapTestFunction(fn);
+        this.name         = name;
+        this.fn           = wrapTestFunction(fn);
+        this.requestHooks = union(this.requestHooks, Array.from(this.fixture.requestHooks));
```
Comment (miherlosev) [inline src/api/structure/test.js]: As I remember, we can use `Array.from(this.fixture.requestHooks)` for cloning request hooks.
```js
this.requestHooks = Array.from(this.fixture.requestHooks);
```

File: test/server/data/test-suites/fixture-is-missing/testfile.js [hunk 2]
```diff
@@ -0,0 +1,6 @@
+// fixture `Fixture`
```
Comment (miherlosev) [inline test/server/data/test-suites/fixture-is-missing/testfile.js]: Should we cover related scenarios?

```js
// fixture `Fixture`

//test('Test', () => {});
```


```js
fixture `Fixture`

//test('Test', () => {});
```

File: src/api/structure/test.js (hunk 1 above)
Comment (AlexKamaev) [reply src/api/structure/test.js]: I'm using it. I just need `union` to merge both `test.requestHooks` and `fixture.request.hooks`

File: test/server/data/test-suites/fixture-is-missing/testfile.js (hunk 2 above)
Comment (AlexKamaev) [reply test/server/data/test-suites/fixture-is-missing/testfile.js]: There will be another error in these cases: NO TESTS to run

Comment (miherlosev) [reply test/server/data/test-suites/fixture-is-missing/testfile.js]: Ok

File: src/api/structure/test.js (hunk 1 above)
Comment (miherlosev) [reply src/api/structure/test.js]: Ok

**Pull request merged**